### DISCUSSION TITLE
Bypassing tasks.json

### DISCUSSION THREAD

**Discussion by adriancostin6** [microsoft/vscode-discussions#519] [Extension Development QnA] [answered]
Is there a way to bypass having to define a debugging `preLaunchTask` in a *tasks.json* file? 

My current use-case is wanting to start a `gdbserver` for debugging in a task that will be run before the actual debug session is started. I would like to do all this in one step once the Debug button is pressed.

Can one register a task inside *extension.ts* and just specify its name as a `preLaunchTask` in *launch.json*?
I have seen https://github.com/microsoft/vscode/issues/128688 but it did not receive enough votes.

My current train of thought is defining all this behavior inside a custom command, but I would like to use the Debug features already in place if possible.

**Comment by isidorn** [comment, marked as answer]
I think this is currently not possible. But @meganrogge can correct me if I am wrong.

**Reply by adriancostin6** [reply]
@isidorn no worries, just wrote a small command that accomplishes what I want instead :)
